lib/image_dart_compiler.dart

```diff
@@ -0,0 +1,47 @@
+import 'package:barback/barback.dart';
+import 'package:path/path.dart';
+import 'src/image_inlining/image_inliner.dart';
+
+const svgExtension = ".svg";
+
+/// Transformer that compiles several image files to a Dart source file with
+/// constant definitions that has these image files Base64-encoded.
+class DartImageCompiler extends AggregateTransformer {
```

Comment:
Given the move out of pub & transformers, you should also (and maybe only) provide a binary entry point + a bazel genrule.

Also, you might consider integrating with the SVG & PNG optimization (as a follow-up).